ds-streamlit/Jenkinsfile.template

```diff
@@ -0,0 +1,73 @@
+// See https://www.opendevstack.org/ods-documentation/ for usage and customization.
+
+@Library('ods-jenkins-shared-library@@ods_git_ref@') _
+
+odsComponentPipeline(
+  imageStreamTag: '@ods_namespace@/jenkins-agent-python:@ods_image_tag@',
+  branchToEnvironmentMapping: [
+    '*': 'dev',
+  ]
+) { context ->
+  createTestVirtualenv(context)
+  stageLinter(context)
+  stageUnitTest(context)
+  odsComponentStageScanWithSonar(context)
+  odsComponentStageBuildOpenShiftImage(context, [
+    buildArgs: [
+      NEXUS_URL_WITH_AUTH: context.nexusHostWithBasicAuth,
+      NEXUS_HOST: context.nexusHostWithoutScheme
+    ],
+    resourceName: "${context.componentId}",
+    dockerDir: "."
+  ])
+  odsComponentStageRolloutOpenShiftDeployment(context)
+}
+
+// ${context.buildUrl} get and set build url in mlflow
+def createTestVirtualenv(def context) {
+  stage('Create virtualenv for tests') {
+    sh(
+      script: """
+        virtualenv testvenv
```

Comment:
I recommend here using the python interpreter calling the venv or virutalenv module, so one can make sure it is using the desired python version (see for example https://github.com/opendevstack/ods-quickstarters/blob/master/be-python-flask/Jenkinsfile.template#L32 )